### PR TITLE
moved GET for user details to own controller

### PR DESCRIPTION
**Pull request by usborn116**
### Routes

Made a /user route to get the current user details, rather than the less-RESTful /get_user route.

### UsersController

Made a separate controller for the new route that only has the 'show' method to get current_user data (if exists) or return false if no current_user

### HomepageController
Removed methods that were unused and/or did not make sense for this particular controller.

### React Components
Replaced all fetch instances of /get_user with /user instead.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class UsersController < ApplicationController
+    
+    def show
+      render json: current_user.as_json(:include => [{:games => {:include => [:sessions]}}, {:sessions => {:include => [:user]}}, {:shared_sessions => {:include => [:user]}}]) || false
+    end
+end
```
Comment (nicholaskillin) [inline app/controllers/users_controller.rb:6]: Nitpicky, but it looks like you are missing an empty line at the end of this file.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class UsersController < ApplicationController
+    
+    def show
+      render json: current_user.as_json(:include => [{:games => {:include => [:sessions]}}, {:sessions => {:include => [:user]}}, {:shared_sessions => {:include => [:user]}}]) || false
```
Comment (nicholaskillin) [inline app/controllers/users_controller.rb:4]: This works, but isn't great to read. I think there are 2 things to consider here:
1. Making this a bit easier to read. We could do something like this:

```suggestion
current_user = User.includes(games: :sessions, sessions: :user, shared_sessions: :user).find(current_user.id)

render json: current_user.as_json(
  include: {
    games: {
      include: :sessions
    },
    sessions: {
      include: :user
    },
    shared_sessions: {
      include: :user
    }
  }
)

      render json: current_user.as_json(:include => [{:games => {:include => [:sessions]}}, {:sessions => {:include => [:user]}}, {:shared_sessions => {:include => [:user]}}]) || false
```
2. With that aside, this isn't very flexible. This makes the assumption that every time you want to fetch the current user, you also always want all of their games, those game sessions, the shared sessions, and so on. With most API's if you sent a request like `GET /user/${userId}`, you would expect to only get the user record back. This API knows too much about what the API wants.

Instead, if you want this additional info you could add `includes` params to your request like this `GET /user/${userId}?include=games,sessions`. This allows the front end to specify what, if any, extra data it needs.

To support this, you would have to modify the controller a bit. Something like this should work (warning: untested code):

```
class UsersController < ApplicationController
  def show
    include_options = build_include_options(params[:include])

    render json: current_user.as_json(include: include_options) || false
  end

  private

  def build_include_options(include_param)
    return {} unless include_param

    include_param.split(',').map do |association|
      case association
      when 'games'
        { games: { include: :sessions } }
      when 'sessions'
        { sessions: { include: :user } }
      when 'shared_sessions'
        { shared_sessions: { include: :user } }
      else
        {}
      end
    end.reduce({}, :merge)
  end
end
```

**Review by nicholaskillin — APPROVED** [review]
This will work, but includes some anti-patterns for building an API. I've made some comments to give you some options, but of course it's up to you if you want to fix them here in this PR or refactor in another one.

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (usborn116) [reply app/controllers/users_controller.rb:4]: oooh good feedback, thanks! I suspect my other controllers suffer from the same issue, so I'll probably do this in a future PR!

**Pull request merged**